example/bench/recipe.cpp

```diff
@@ -0,0 +1,54 @@
+#include <random>
+
+#include <benchmark_cell.hpp>
+#include <common_types.hpp>
+#include <time_sequence.hpp>
+
+#include "recipe.hpp"
+
+cell_size_type bench_recipe::num_cells() const {
+    return params_.num_cells;
+}
+
+arb::util::unique_any bench_recipe::get_cell_description(cell_gid_type gid) const {
+    using RNG = std::mt19937_64;
+    arb::benchmark_cell cell;
+    cell.realtime_ratio = params_.cell.realtime_ratio;
+    cell.time_sequence = arb::poisson_time_seq<RNG>(RNG(gid), 0, 1e-3*params_.cell.spike_freq_hz);
+    return std::move(cell);
+}
+
+arb::cell_kind bench_recipe::get_cell_kind(arb::cell_gid_type gid) const {
+    return arb::cell_kind::benchmark;
+}
+
+std::vector<arb::cell_connection> bench_recipe::connections_on(cell_gid_type gid) const {
+    const auto n = params_.network.fan_in;
+    std::vector<arb::cell_connection> cons;
+    cons.reserve(n);
+    using RNG = std::mt19937_64;
+    RNG rng(gid);
+    // issue: invalid gid for source in a connection
```

Comment:
Don't understand this comment

Reply:
Sorry, this was a remnant from debugging.
It has been replaced with the following comment:
```C++
// Generate n incoming connections on this cell with random source, where
// the source can't equal gid (i.e. no self-connections).                                                                                                                                                           
// We use a random distribution that will uniformly draw values from the 
// union of the two ranges: [0, gid-1] AND [gid+1, num_cells-1].
// To do this, we draw random values in the range [0, num_cells-2], then
// add 1 to random values ≥ gid.
```
Which should help with your comment below about why `src` is incremented when greater than or equal to `gid`.